pkg/services/auth/jwt/auth_test.go

```diff
@@ -465,3 +664,43 @@ func configurePKIXPublicKeyFile(t *testing.T, cfg *setting.Cfg) {
 
 	cfg.JWTAuth.KeyFile = file.Name()
 }
+
+// Copied from pkg/setting/setting_secure_socks_proxy_test.go and modified key size for faster generation
+func createTestRootCAFile(t *testing.T, path string) string {
+	t.Helper()
+
+	ca := &x509.Certificate{
+		SerialNumber: big.NewInt(2019),
+		Subject: pkix.Name{
+			Organization: []string{"Grafana Labs"},
+			CommonName:   "Grafana",
+		},
+		NotBefore:             time.Now(),
+		NotAfter:              time.Now().AddDate(10, 0, 0),
+		IsCA:                  true,
+		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth, x509.ExtKeyUsageServerAuth},
+		KeyUsage:              x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
+		BasicConstraintsValid: true,
+	}
+	caPrivKey, err := rsa.GenerateKey(rand.Reader, 1024) //nolint:gosec
+	require.NoError(t, err)
+	caBytes, err := x509.CreateCertificate(rand.Reader, ca, ca, &caPrivKey.PublicKey, caPrivKey)
+	require.NoError(t, err)
+
+	// nolint:gosec
+	caCertFile, err := os.Create(path)
+	require.NoError(t, err)
+
+	block := &pem.Block{
+		Type:  "CERTIFICATE",
+		Bytes: caBytes,
+	}
+	err = pem.Encode(caCertFile, block)
+	require.NoError(t, err)
+
+	buf := new(bytes.Buffer)
+	err = pem.Encode(buf, block)
+	require.NoError(t, err)
+
+	return buf.String()
+}
```

Comment:
It might be slightly cleaner to leverage `testcerts` here (we're using it in other places as well):

```suggestion
func createTestRootCAFile(t *testing.T) string {
	t.Helper()

	tmpDir := t.TempDir()
	ca := testcerts.NewCA()
	caCertFile, _, err := ca.ToTempFile(tmpDir)
	require.NoError(t, err)

	return caCertFile.Name()
}
```

Then you can update the callsite to stop providing the name and use the returned name instead.

Reply:
I was searching the codebase for other places where test certificates were created, unfortunately I found the other one first as this is quite a lot cleaner!

Reply:
Since `createTestRootCAFile` is almost selfcontained now, should the cleanup function be included?
```go
t.Cleanup(func() {
	if err := os.Remove(caCertFile.Name()); err != nil {
		panic(err)
	}
})
```

Reply:
Good question. I suppose we can either way. I don't feel strongly about either approach. I'd probably even not worry about cleanup at all, haha. I think this is fine as-is.

Reply:
I'll leave it be as is then.